ext/opentelemetry-ext-flask/tests/test_automatic.py

```diff
@@ -0,0 +1,98 @@
+
+from opentelemetry import trace
+from opentelemetry.configuration import Configuration
+from opentelemetry.ext.flask import FlaskInstrumentor
+from opentelemetry.test.wsgitestutil import WsgiTestBase
+
+# pylint: disable=import-error
+from .base_test import InstrumentationTest, expected_attributes
+
+
+class TestAutomatic(WsgiTestBase, InstrumentationTest):
+    def setUp(self):
+        super().setUp()
+
+        Configuration._instance = None  # pylint: disable=protected-access
+        Configuration.__slots__ = []  # pylint: disable=protected-access
+        FlaskInstrumentor().instrument()
+
+        self.app = flask.Flask(__name__)
+
+        def hello_endpoint(helloid):
+            if helloid == 500:
+                raise ValueError(":-(")
+            return "Hello: " + str(helloid)
+
+        def excluded_endpoint():
+            return "excluded"
+
+        def excluded2_endpoint():
+            return "excluded2"
+
+        self.app.route("/hello/<int:helloid>")(hello_endpoint)
+        self.app.route("/excluded/<int:helloid>")(hello_endpoint)
+        self.app.route("/excluded")(excluded_endpoint)
+        self.app.route("/excluded2")(excluded2_endpoint)
+
+        self.client = Client(self.app, BaseResponse)
+
+    def tearDown(self):
+        disable(WARNING)
+        FlaskInstrumentor().uninstrument()
+        disable(NOTSET)
+
+    def test_uninstrument(self):
+        expected_attrs = expected_attributes(
+            {"http.target": "/hello/123", "http.route": "/hello/<int:helloid>"}
+        )
+        resp = self.client.get("/hello/123")
+        self.assertEqual(200, resp.status_code)
+        self.assertEqual([b"Hello: 123"], list(resp.response))
+        span_list = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(span_list), 1)
+        self.assertEqual(span_list[0].name, "hello_endpoint")
+        self.assertEqual(span_list[0].kind, trace.SpanKind.SERVER)
+        self.assertEqual(span_list[0].attributes, expected_attrs)
+
+        FlaskInstrumentor().uninstrument()
+        self.app = flask.Flask(__name__)
+
+        def hello_endpoint(helloid):
+            if helloid == 500:
+                raise ValueError(":-(")
+            return "Hello: " + str(helloid)
+
+        self.app.route("/hello/<int:helloid>")(hello_endpoint)
+
+        self.client = Client(self.app, BaseResponse)
+
+        expected_attrs = expected_attributes(
+            {"http.target": "/hello/123", "http.route": "/hello/<int:helloid>"}
+        )
+        resp = self.client.get("/hello/123")
+        self.assertEqual(200, resp.status_code)
+        self.assertEqual([b"Hello: 123"], list(resp.response))
+        span_list = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(span_list), 1)
+        self.assertEqual(span_list[0].name, "hello_endpoint")
+        self.assertEqual(span_list[0].kind, trace.SpanKind.SERVER)
+        self.assertEqual(span_list[0].attributes, expected_attrs)
```

Comment:
These values are generated before the call to `uninstrument`, why to check them again?
Just checking that no new spans were generated should be enough.

Reply:
Fixed